### ISSUE TITLE
Detect Enter Key Press to Proceed 

### ISSUE THREAD

**Issue by GovindKrishnan** [AmruthPillai/Reactive-Resume#751]
It's user friendly to detect **Enter Key Press to Proceed in Dialogs** [Eg: Create Resume] across the site. 

Hampers speed when you have to click the buttons everytime.

**Comment by chandiwalaaadhar** [comment]
Great Idea, I’ll try to take this up and raise a PR

**Comment by GovindKrishnan** [comment]
> Great Idea, I’ll try to take this up and raise a PR

😃👍

**Comment by chandiwalaaadhar** [comment]
I couldn't really decide on the UX because there are text fields in all the modals and I think pressing "Enter" key has to move to new line as it is doing currently. @AmruthPillai any suggestions?

**Comment by GovindKrishnan** [comment]
> I couldn't really decide on the UX because there are text fields in all the modals and I think pressing "Enter" key has to move to new line as it is doing currently. @AmruthPillai any suggestions?

@chandiwalaaadhar, the general standard followed universally is `Tab` for next field, `Esc` for cancel & `Enter` for submit.

`Tab` & `Esc` work as intended already.

**Comment by chandiwalaaadhar** [comment]
> > I couldn't really decide on the UX because there are text fields in all the modals and I think pressing "Enter" key has to move to new line as it is doing currently. @AmruthPillai any suggestions?
> 
> @chandiwalaaadhar, the general standard followed universally is `Tab` for next field, `Esc` for cancel & `Enter` for submit.
> 
> `Tab` & `Esc` work as intended already.

That’s right @GovindKrishnan but we have a text box where we mention the description of our work, and we would not want to restrict the user there such that pressing enter leads to submission of the form

**Comment by GovindKrishnan** [comment]
> > > I couldn't really decide on the UX because there are text fields in all the modals and I think pressing "Enter" key has to move to new line as it is doing currently. @AmruthPillai any suggestions?
> > 
> > @chandiwalaaadhar, the general standard followed universally is `Tab` for next field, `Esc` for cancel & `Enter` for submit.
> > 
> > `Tab` & `Esc` work as intended already.
> 
> That’s right @GovindKrishnan but we have a text box where we mention the description of our work, and we would not want to restrict the user there such that pressing enter leads to submission of the form

True. Let's see what @AmruthPillai got to say.

**Comment by AmruthPillai** [comment]
This was implemented in modals previously, but I believe I ran into the same issue where I was not able to exclude key presses from within the text area input.

This is priority number one in terms of good UX. Will see what would be the best practice to add enter submit keypress to all modals, with minimal amount of code (probably by adding it inside BaseModal)

**Comment by AmruthPillai** [comment]
This issue was automatically closed as it was created before 2024-01-01.
